tests/src/test/scala/cilib/BoundaryTest.scala

```diff
@@ -0,0 +1,69 @@
+package cilib
+
+import scalaz.NonEmptyList
+import org.scalacheck._
+import org.scalacheck.Prop._
+import org.scalacheck.Gen._
+import org.scalacheck.Arbitrary._
+
+import spire.math.Interval
+import spire.implicits._
+
+object BoundaryTest extends Spec("Boundary enforcement") {
+
+  final case class Foo(b: Interval[Double], a: Double)
+
+  val doubleGen = Gen.choose(-1000000.0, 1000000.0)
+
+  val intervalGen: Gen[Interval[Double]] =
+      for {
+        a <- doubleGen
+        b <- doubleGen suchThat (_ > a)
+      } yield Interval(a, b)
+  implicit def arbInterval = Arbitrary { intervalGen }
+
+  implicit def fooGen = Arbitrary {
+    for {
+      i <- intervalGen
+      range = math.abs(i.upperValue - i.lowerValue) * 0.4
```

Comment:
Using ```0.4``` never generates the breaking example I gave above if ```b=(5,6)``` and ```a=3```, this case still breaks the equation.

Reply:
I think it needs to be at least 2 times the range

Reply:
This is a fundamental issue with this boundary mechanism. Do we just keep reflecting until the value is in range? Does that even make sense? I've not seen such an adjustment in anything that I've read so far.

Reply:
Hehe, I guess you can keep reflecting, it's similar to what the periodic one does with the mod.

Reply:
Yeah, that's the only thing that could possibly work... seems a little strange though. I guess it can't be too bad then, because the point of the schemes to to bring something back into the value space. They _all_ make the assumption that the initial starting point is within the bounds.